templates/nginx-maintenance.conf

```
@@ -0,0 +1,54 @@
+server {
+    listen 127.0.0.1:{{ MAINTENANCE_SERVER_PORT }};
```

Instead of listening on different ports, we could also use different paths, e.g. `/error`, `/maintenance` etc.  I'm not requesting a change here, just suggesting an alternative that may be simpler.

---

The issue I had with that was that if you override the backend during maintenance, or if all AppServers are inactive, then it is using the error/maintenance backend and not redirecting to a URL. For instance here: https://pr17332.sandbox.opencraft.hosting/. 
Additionally, I wasn't clear about whether maintenance (non-error) pages should throw a 50x error (they don't in this PR) when it is (un)scheduled.

---

I think that both scheduled and unscheduled maintenance pages should return a "503 Service Unavailable" code. That's what that code is for.

The error backend should return the original 5xx error if possible, or otherwise 500 if we have to choose.

---

@bradenmacdonald Sounds reasonable.  We will immediately be paged if we set up the manitenance page, but I guess that's a good thing, since we will also be paged if the maintenance page apppears due to a bug when we didn't plan it.  We can still figure out other ways of silencing the pager during a planned maintenance.

---

@xitij2000 OK, fair enough, it's fine with different ports.